### PR TITLE
start unifying controller for consistent html/json/csv behavior

### PR DESCRIPTION
**Pull request by grosser**
as replacement for https://github.com/zendesk/samson/pull/3150 and as future basis for most of the controllers

TODO:
 - use it for more controllers (commands/stages/deploys)
 - add more test coverage

### CODE REVIEW

**Comment by grosser** [comment]
@danihodovic

**Comment by danihodovic** [comment]
:clap: :clap: 

What do you think about applying resource_controller.rb to stages and commands to see if it's as generic as it should be?

**Comment by danihodovic** [comment]
Great :+1: . This will simplify building tooling around Samson since the API is now restful-ish. cc @elblivion @tolgaakyuz.

File: test/controllers/resource_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    json.keys.must_equal [:project]
+  end
+
+  def json
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  let(:project) { projects(:test) }
+
+  as_a :admin do
```
Comment (bcolferzd) [inline test/controllers/resource_controller_test.rb]: Do you need "as_a :user" or "as_not_a :admin" ?

**Comment by dragonfax** [comment]
Its just a cosmetic test failure.

File: test/controllers/resource_controller_test.rb (hunk 1 above)
Comment (grosser) [reply test/controllers/resource_controller_test.rb]: need admin login here so all urls work like deleting commands etc

**Comment by danihodovic** [comment]
Any plans on applying this for stages and deploy groups?

**Comment by grosser** [comment]
all controller should get these, I'm doing them 1-by-1 when I have a bit of time to kill :D

**Pull request merged**